impedance/models/circuits/fitting.py

```diff
@@ -148,6 +153,10 @@ def circuit_fit(frequencies, impedances, circuit, initial_guess, constants={},
         if weight_by_modulus:
             abs_Z = np.abs(Z)
             kwargs['sigma'] = np.hstack([abs_Z, abs_Z])
+        elif weight_proportionally and not weight_by_modulus:
+            Re_Z = np.real(Z)
+            Im_Z = np.imag(Z)
+            kwargs['sigma'] = np.hstack([Re_Z, Im_Z])
```

Comment:
Since this is a complex number you can make this clearer (in my mind at least) by using '.real' and '.imag'

```suggestion
            kwargs['sigma'] = np.hstack([Z.real, Z.imag])
```

Reply:
The other consideration here is whether you want to weight by the absolute value of the real (doesn't matter very much since almost always positive) and imaginary (definitely matters as both positive and negative values will exist) and not just the values themselves. Not sure how the weight argument is interpreted